ai_graph_color/algorithms/backtracking.py

```diff
@@ -0,0 +1,51 @@
+def run(setup, problem, *params):
+    """
+    This will execute the algorithm
+
+    :param setup: the setup for this algorithm
+    :type setup: :class: ~ai_graph_color.Setup
+    :param problem: the graph to color
+    :type problem: the graph to color
+    :param params: the params for this algorithm
+    :type params: tuple
+    :rtype: list[int] the list of coloring
+    :return: the list of colorings for this graph
+    """
+    len_problem = 0
+    for adj_list in problem:
+        len_problem += len(adj_list)
```

Comment:
So, you're finding 2 * the number of connections in the graph? I think the "length of the problem" is just the number of nodes right?